wallet-new/docs/spec.tex

```diff
@@ -3009,38 +3025,533 @@ \subsection{Use cases}
+a situation like this (and indeed, exchanges have additional infrastructure in
+place that alleviates this problem).
+
+\subsection{Conclusions}
+
+The choice of coin selection algorithm has far reaching consequences on the long
+term behaviour of a cryptocurrency wallet. To a large extent the coin selection
+algorithm determines, over time, the shape of the UTxO. Moreover, the performance
+of the algorithm can be of crucial importance to high-traffic wallets such as
+exchange nodes.
+
+In his thesis, Erhardt proposes ``Branch and Bound'' as his preferred coin
+selection algorithm. Branch and Bound in essense is a limited backtracking
+algorithm that tries to find an exact match, so that no change output needs
+to be generated at all. When the backtracking cannot find an exact match within
+its bounds, the algorithm then falls back on random selection. It does not,
+however, attempt our ``improvement'' step, and instead just attempts to reach
+a minimum but fixed change size, to avoid generating dust. It is hard to compare
+the two algorithms directly, but on the MoneyPot dataset at least the results
+are comparable; Erhardt ends up with a slightly smaller average UTxO (109
+versus our 130), and a slightly smaller average number of inputs (2.7 versus
+our 3.0). In principle we could modify our \texttt{Random-Improve} algorithm to
+start with bounded backtracking to find an exact match, just like Erhardt does;
+we have not done this however because it adds complexity to the algorithm and
+reduces performance. Erhardt reports that his algorithm is able to find
+exact matches in 30\% of the time. This is \emph{very} high, and at least partly
+explains why his UTxO and average number of change outputs is lower; in the
+Cardano blockchain, we would not expect that there \emph{exist} exact matches
+anywhere near that often (never mind \emph{finding} them).
+
+Instead our proposed \texttt{Random-Improve} does no search at all, instead purely
+relying on self organization principles, the first of which was stated by
+Erhardt, and the other two we identified as part of this research. Although
+in the absence of more real data it is hard to evaluate any coin selection
+algorithm, we have shown that the algorithm performs well across a large
+variety of different distributions and deposit:payment ratios. Moreover it is
+straight-forward to implement and has high performance.
+
+One improvement we may wish to consider is that \emph{when} there are very large
+deposits, we could occassionally issue a ``reorganization transaction'' that
+splits those large deposits into smaller chunks. This would bring the
+change:payment ratio down, which would improve the evolution of the UTxO over
+time and is beneficial also for other, more technical reasons (it reduces the
+need for dependent transactions). Such reorganization is largely orthogonal to
+this algorithm, however, and can be implemented independently.
```

Comment:
Yes, I think we should add a sentence or two earlier on this. We don't need to run more sims but we already have a fairly good idea of what to do and what we would expect.